File: src/main/java/com/karan/restaurant/review/system/service/ReviewService.java
package com.karan.restaurant.review.system.service;

import com.karan.restaurant.review.system.model.Category;
import com.karan.restaurant.review.system.model.Review;

import java.util.List;

public interface ReviewService {
    public Review addReview(Review review, String restaurantId, String userId);
    public Review updateReview(Long id, Review review);
    public void deleteReview(Long id);
    public Review getReviewById(Long id);
    public List<Review> getAllReviews();
}

File: frontend/src/services/reviewService.js
import api from '../utils/api';

export const reviewService = {
  // Create review
  createReview: async (reviewData, restaurantId, userId) => {
    const response = await api.post(`/reviews/add?restaurantId=${restaurantId}&userId=${userId}`, reviewData);
    return response.data;
  },

  // Get all reviews
  getAllReviews: async () => {
    const response = await api.get('/reviews');
    return response.data;
  },

  // Get review by ID
  getReviewById: async (id) => {
    const response = await api.get(`/reviews/${id}`);
    return response.data;
  },

  // Update review
  updateReview: async (id, reviewData) => {
    const response = await api.put(`/reviews/${id}`, reviewData);
    return response.data;
  },

  // Delete review
  deleteReview: async (id) => {
    await api.delete(`/reviews/${id}`);
  },
};